website/content/en/docs/building-operators/golang/project_migration_guide.md

```diff
@@ -1,182 +1,315 @@
+The Memcached API types will look like:
 
 ```go
+// MemcachedSpec defines the desired state of Memcached
 type MemcachedSpec struct {
-	// Specify size, the number of pods
+	// Size is the size of the memcached deployment
 	Size int32 `json:"size"`
 }
 
+// MemcachedStatus defines the observed state of Memcached
 type MemcachedStatus struct {
 	// Nodes are the names of the memcached pods
+	// +listType=set
 	Nodes []string `json:"nodes"`
 }
+
+// +kubebuilder:object:root=true
+
+// Memcached is the Schema for the memcacheds API
+type Memcached struct {...}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// MemcachedList contains a list of Memcached
+type MemcachedList struct {...}
 ```
-**Note**:
-If there are any libraries or pkgs present in `pkg/apis/cache/v1alpha1`, copy them over to `api/v1alpha1`.
 
-### CRD markers
+### Migrating the Controllers
 
-Include any of the CRD generation or validation markers defined for your API type and fields in `api/<version>/<kind>_types.go`.
+In our example, we will migrate the code from `pkg/controller/memcached/memcached_controller.go` to `controllers/memcached_controller.go`. 
 
-For `memcached-operator` the following markers present in `pkg/apis/cache/v1aplha1/memcached_types.go` are copied to `api/v1alpha1/memcached_types.go`.
+Now, the struct Reconcile<Kind> is called <KindReconciler>, so if you have customizations copy the fields from the, 
+for example, `ReconcileMemcached` struct to `MemcachedReconciler`.
 
-Example:
-```Go
-// +kubebuilder:subresource:status
-// +kubebuilder:resource:path=memcacheds,scope=Namespaced
+We’ll need to replace the `// your logic here` in the new layout for our reconcile logic. See that in our example the:
+
+```
+func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+```
+
+Is replaced with:
+
+```
+func (r *MemcachedReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+```
+
+Note that:
+
+- The **Log** in the new layout has not a variable to be used. In this way, replace:
+
+```go
+_ = r.Log.WithValues("memcached", req.NamespacedName)`
 ```
 
-### Migrate controller
+With:
 
-Copy over the `Reconcile()` code and the controller logic from your existing controller at `pkg/controller/<kind>/<kind>_controller.go` to the new controller's reconcile function in Kubebuilder project at `controllers/<kind>_controller.go`.
+```go
+log := r.Log.WithValues("memcached", req.NamespacedName)
+```
 
-In our example [`pkg/controller/memcached/memcached_controller.go`][memcached_controller] is copied to `controllers/memcached_controller.go`.
+And then, you we use `log` instead of `reqLogger`. Example:
+
+```go
+log.Info("Memcached resource not found. Ignoring since object must be deleted")
+```
 
-Particularly, note the following changes:
+- The `r.client.Action` will be replaced with `r.Action`. (e.g replace `r.client.Get` with `r.Get`)
+- The `r.scheme` will be replaced with `r.Scheme`
```

Comment:
`Copy all existing Reconcile<Kind> code into the new controller.go file, overwriting the existing Reconcile implementation.` and they try to fix shows pitiful since the imports will not be the same as other nits details. also, they might end up to changing properly the watches implementation. 

I agree that it is not good I will try to come up with a better explanation.